src/core/ActionAtomistic.cpp

```diff
@@ -77,7 +77,8 @@ void ActionAtomistic::requestAtoms(const vector<AtomNumber> & a) {
 // only real atoms are requested to lower level Atoms class
     else unique.insert(indexes[i]);
   }
-
+  updateUniqueLocal();
+  atoms.unique.clear();
```

Comment:
@carlocamilloni why is it necessary to clear the atoms.unique set at this point? I've seen a number of unique.clear() around that I did not understand